apps/sms/services/js/moz_mobile_connections/moz_mobile_connections_shim.js

```diff
@@ -0,0 +1,106 @@
+/* global BridgeServiceMixin,
+          BroadcastChannel,
+          MozSettingsShim
+*/
+
+/* exported MozMobileConnectionsShim */
+
+(function(exports) {
+'use strict';
+
+/**
+ * Name of the service for mozMobileConnections API shim.
+ * @type {string}
+ */
+const SERVICE_NAME = 'moz-mobile-connections-shim';
+
+/**
+ * Array of method names that need to be exposed for API shim.
+ * @type {Array.<string>}
+ */
+const METHODS = Object.freeze(['getServiceIdByIccId', 'switchMmsSimHandler']);
+
+var mozMobileConnections = null;
+
+function setMmsSimServiceId(id) {
+  // DSDS: mms & data are both necessary for connection switch.
+  MozSettingsShim.set({
+    'ril.mms.defaultServiceId': id,
+    'ril.data.defaultServiceId': id
+  });
+}
+
+var MozMobileConnectionsShim = {
+  init(appInstanceId, conns) {
+    if (!conns) {
+      console.error('Invalid mozMobileConnections for shim initialization');
+      return;
+    }
+
+    var broadcastChannelName = `${SERVICE_NAME}-channel-${appInstanceId}`;
```

Comment:
__optional nit:__ maybe no need in separate variable for ```broadcastChannelName``` here, I was using separate variable in other shim, just because I was logging it with ```debug```, but you don't do it anyway :)